package/MDAnalysis/auxiliary/XVG.py

```diff
@@ -0,0 +1,245 @@
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+XVG auxiliary reader --- :mod:`MDAnalysis.auxiliary.XVG`
+========================================================
+
+.. autofunction:: uncomment
+
+.. autoclass:: XVGReader
+   :members:
+
+.. autoclass:: XVGFileReader
+   :members:
+
+"""
+
+import os
+import numpy as np
+from . import base
+
+def uncomment(lines):
+    """ Remove comments from lines in an .xvg file
+
+    Parameters
+    ----------
+    lines : list of str
+        Lines as directly read from .xvg file
+
+    Yields
+    ------
+    str
+        The next non-comment line, with any trailing comments removed
+    """
+    for line in lines:
+        # '@' must be at the beginning of a line to be a grace instruction
+        if line.lstrip()[0] == '@':
+            continue
+        # '#' can be anywhere in the line, everything after is a comment
+        comment_position = line.find('#')
+        if comment_position > 0 and line[:comment_position]:
+            yield line[:comment_position]
+        elif comment_position < 0 and line:
+            yield line
+        # if comment_position == 0, then the line is empty
+
+
+class XVGReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .xvg file.
+
+    All data from the file will be read and stored on initialisation. This is
+    the default reader for .xvg files.
+    
+    xvg files are produced by Gromacs during simulation or analysis, formatted
+    for plotting data with Grace.
+
+    Parameters
+    ----------
+    filename : str
+       Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.    
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+
+    Note
+    ----
+    The time of each step is assumed to stored in the first column (``time_col``
+    defaults to 0), in units of ps, and data is assumed to be time-ordered.
+
+    The file is assumed to be of a size such that reading and storing the full 
+    contents is practical.
+    """
+
+    format = "XVG"
+
+    def __init__(self, filename, **kwargs):
+        self._data_input = os.path.abspath(filename)
+        with open(filename) as xvg_file:
+            lines = xvg_file.readlines()
+        auxdata = []
+        # remove comments before storing
+        for i, line in enumerate(uncomment(lines)):
+           auxdata.append([float(l) for l in line.split()])
+           # check the number of columns is consistent
+           if len(auxdata[i]) != len(auxdata[0]):
+                raise ValueError('Step {0} has {1} columns instead of '
+                                 '{2}'.format(i, auxdata[i], auxdata[0]))
+        self._auxdata = np.array(auxdata)
+        # default to time in first column, if not otherwise specified
+        time_col = kwargs.pop('time_col', 0)

```

Comment:
the solution from @jbarnoud is what I implied. I'm ok passing kwargs to the baseclass. But we shouldn't pop from them. Any argument used by the init function directly should be named explicitly in the definition. This means the default will show up in the docs and the ipython function introspection can also show me the default straight away.